system/include/emscripten/threading.h

```diff
@@ -36,7 +36,8 @@ void emscripten_force_num_logical_cores(int cores);
 
 // If the given memory address contains value val, puts the calling thread to
 // sleep waiting for that address to be notified.
-int emscripten_futex_wait(volatile void/*uint32_t*/ *addr __attribute__((nonnull)), uint32_t val, double maxWaitMilliseconds);
+// Returns -EINVAL if addr is null.
+int emscripten_futex_wait(volatile void/*uint32_t*/ * _Nonnull addr, uint32_t val, double maxWaitMilliseconds);
```

Comment:
Out if interest what would happen if we just used `uint32_t` here?  (rather than void with the comment).

Reply:
The `void *` is there to enable the use of untyped C typing rule that pointer to anything is allowed to implicitly cast to a pointer to void.

This is following the GCC atomics intrinsics semantics where atomic operations on an address are allowed to work on a pointer to any type, e.g. https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html

Otherwise if this was
```
emscripten_futex_wait(volatile uint32_t *addr, ...);
```
then none of
```
int x;
signed int y;
unsigned int z;
int32_t w;
emscripten_futex_wait(&x, ...);
emscripten_futex_wait(&y, ...);
emscripten_futex_wait(&z, ...);
emscripten_futex_wait(&w, ...);
```
would compile, but one would have to always write
```
emscripten_futex_wait((uint32_t*)&x, ...);
```

In hindsight, this compatibility was maybe not that important, but that was the rationale iirc.